rust/datafusion/src/logicalplan.rs

```diff
@@ -355,7 +355,7 @@ pub enum Expr {
         /// List of expressions to feed to the functions as arguments
         args: Vec<Expr>,
         /// The `DataType` the expression will yield
-        return_type: DataType,
+        return_type: ReturnType,
```

Comment:
THIS IS A MAJOR CHANGE: it makes Expr hold an `Arc<&dyn>`, which make them unserializable. I was unable to find another way, unfortunately.